apps/namadillo/src/App/Masp/MaspShield.tsx

```diff
@@ -127,38 +115,30 @@ export const MaspShield: React.FC = () => {
         chainId,
       });
 
-      const txResponse = await performShieldTransfer.mutateAsync({
-        sourceAddress,
-        destinationAddress,
-        tokenAddress: selectedAsset.originalAddress,
-        amount: displayAmount,
-        gasConfig,
-      });
+      const txResponse = await performTransfer();
 
-      // TODO review and improve this data to be more precise and full of details
-      const tx: TransferTransactionData = {
-        type: "TransparentToShielded",
-        currentStep: TransferStep.Complete,
-        sourceAddress,
-        destinationAddress,
-        asset: selectedAsset.asset,
-        displayAmount,
-        rpc: txResponse.msg.payload.chain.rpcUrl,
-        chainId: txResponse.msg.payload.chain.chainId,
-        hash: txResponse.encodedTxData.txs[0]?.hash,
-        feePaid: txResponse.encodedTxData.wrapperTxProps.feeAmount,
-        resultTxHash: txResponse.encodedTxData.txs[0]?.innerTxHashes[0],
-        status: "success",
-        createdAt: new Date(),
-        updatedAt: new Date(),
-      };
-      setTransaction(tx);
-      storeTransaction(tx);
-
-      setCurrentStepIndex(2);
+      if (txResponse) {
+        const txList = createTransferDataFromNamada(
+          txKind,
+          selectedAsset.asset,
+          rpcUrl,
+          txResponse
+        );
+
+        // Currently we don't have the option of batching transfer transactions
+        if (txList.length === 0) {
+          throw "Couldn't create TransferData object ";
+        }
+
+        const tx = txList[0];
+        setTransaction(tx);
+        storeTransaction(tx);
+      } else {
+        throw "Invalid transaction response";
+      }
     } catch (err) {
       setGeneralErrorMessage(err + "");
-      setCurrentStepIndex(0);
+      setTransaction(undefined);
```

Comment:
After talking to @ChrisHoltDesign we won't have the timeline active on the "Signing" anymore, which simplifies A LOT of things. Maybe you can redirect directly to the transactions page after the `storeTransaction` just like I did on #1454  - we can do it in a later PR too

Reply:
Oh, this is a copy and paste from <NamadaTransfer> that you implemented 😁 

I'll open a new PR focused on the Timeline 👍